addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import re
+from datetime import timedelta
+
+import requests
+from requests.exceptions import ConnectTimeout, HTTPError, ReadTimeout
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductFetchImageWizard(models.TransientModel):
+    _name = 'product.fetch.image.wizard'
+    _description = "Fetch product images from Google Images based on the product's barcode number."
+
+    @api.model
+    def default_get(self, fields_list):
+        # Check if the cron is already triggered and raise an UserError if so
+        cron_id = self.env.ref('product_images.ir_cron_fetch_image').id
+        date = self.env['ir.cron.trigger'].search(
+            [('cron_id', '=', cron_id)], limit=1, order="call_at desc"
+        ).call_at
+        if date:
+            raise UserError(_(
+                "A task is already scheduled to pick the images. Please try again later."
+            ))
+
+        # Check if API keys are set
+        ICP = self.env['ir.config_parameter']
+        google_pse_id = bool(ICP.get_param('google.pse.id'))
+        google_custom_search_key = bool(ICP.get_param('google.custom_search.key'))
+        if not google_pse_id or not google_custom_search_key:
```

Comment:
```suggestion
        if not (google_pse_id and google_custom_search_key):
```
is often more readable (YMMV though)